app/src/main/java/com/connfa/service/firebase/FirebaseConnfaRepository.java

```diff
@@ -0,0 +1,115 @@
+package com.connfa.service.firebase;
+
+import android.content.Context;
+
+import com.connfa.R;
+import com.connfa.service.api.ConnfaRepository;
+import com.connfa.service.api.ConnfaService;
+import com.connfa.service.firebase.model.FirebaseEvent;
+import com.connfa.service.firebase.model.FirebaseFloorPlan;
+import com.connfa.service.firebase.model.FirebaseInfoItem;
+import com.connfa.service.firebase.model.FirebaseLevel;
+import com.connfa.service.firebase.model.FirebaseLocation;
+import com.connfa.service.firebase.model.FirebasePoi;
+import com.connfa.service.firebase.model.FirebaseSettings;
+import com.connfa.service.firebase.model.FirebaseSpeaker;
+import com.connfa.service.firebase.model.FirebaseTrack;
+import com.connfa.service.firebase.model.FirebaseType;
+import com.connfa.service.model.Updates;
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseError;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+import com.google.firebase.database.ValueEventListener;
+import com.jakewharton.retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory;
+
+import java.util.Arrays;
+
+import io.reactivex.Observable;
+import io.reactivex.ObservableEmitter;
+import io.reactivex.ObservableOnSubscribe;
+import io.reactivex.functions.Function;
+import io.reactivex.schedulers.Schedulers;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+public final class FirebaseConnfaRepository {
+    private final DatabaseReference database;
+
+    public static FirebaseConnfaRepository newInstance(Context context) {
+        DatabaseReference database = FirebaseDatabase.getInstance().getReference();
+        return new FirebaseConnfaRepository(database);
+    }
+
+    private FirebaseConnfaRepository(DatabaseReference database) {
+        this.database = database;
+    }
+
+    public Observable<FirebaseEvent.Holder> bofs() {
```

Comment:
Me too, I didn't find anything in the web. Maybe we should go to a Drupal Conf to know 😆